app/sync.js

```diff
@@ -250,10 +254,20 @@ module.exports.onSyncReady = (isFirstRun, e) => {
  */
 module.exports.init = function (initialState) {
   const RELOAD_MESSAGE = 'reload-sync-extension'
+  const RESET_SYNC = 'reset-sync'
   // sent by about:preferences when sync should be reloaded
   ipcMain.on(RELOAD_MESSAGE, () => {
     process.emit(RELOAD_MESSAGE)
   })
+  // sent by about:preferences when resetting sync
+  ipcMain.on(RESET_SYNC, (e) => {
+    if (dispatcherCallback) {
+      dispatcherCallback({actionType: syncConstants.SYNC_DELETE_USER})
+    }
+    appActions.changeSetting(settings.SYNC_ENABLED, false)
```

Comment:
i would wait a short amount of time to call this to make sure that SYNC_DELETE_USER gets processed before the extension is disabled